frost-core/src/frost/round2.rs

```diff
@@ -187,6 +187,15 @@ pub fn sign<C: Ciphersuite>(
     signer_nonces: &round1::SigningNonces<C>,
     key_package: &frost::keys::KeyPackage<C>,
 ) -> Result<SignatureShare<C>, Error<C>> {
+    // Validate the signer's commitment is present in the signing package
+    if signing_package
+        .signing_commitments
+        .get(&key_package.identifier)
+        .is_none()
+    {
+        return Err(Error::MissingCommitment);
+    }
```

Comment:
As discussed in chat, check if the commitment matches the one derived from `signer_nonces`

Reply:
Done in [77e7510](https://github.com/ZcashFoundation/frost/pull/452/commits/77e7510d4c9c650fd9e3f6279b20e666ef634e6e)